node/src/accountant/mod.rs

```diff
@@ -3381,429 +4099,450 @@ mod tests {
 
-        TestLogHandler::new().exists_log_containing(&format!(
-            "ERROR: Accountant: Overflow error recording consumed services from {}: total charge {}, service rate {}, byte rate 1, payload size 2. Skipping",
-            wallet,
-            i64::MAX as u64 + 1 * 2,
-            i64::MAX as u64
-        ));
+        System::current().stop();
+        assert_eq!(system.run(), 0);
+        let mut finish_scan_params = finish_scan_params_arc.lock().unwrap();
+        let (actual_sent_payable, logger) = finish_scan_params.remove(0);
+        assert_eq!(actual_sent_payable, sent_payable,);
+        test_use_of_the_same_logger(&logger, test_name, None);
+        let mut payable_notify_later_params = payable_notify_later_params_arc.lock().unwrap();
+        let (scheduled_msg, _interval) = payable_notify_later_params.remove(0);
+        assert_eq!(scheduled_msg, ScanForNewPayables::default());
+        assert!(
+            payable_notify_later_params.is_empty(),
+            "Should be empty but {:?}",
+            payable_notify_later_params
+        );
     }
 
     #[test]
-    #[should_panic(
-        expected = "Recording services consumed from 0x000000000000000000000000000000626f6f6761 but \
-     has hit fatal database error: RusqliteError(\"we cannot help ourselves; this is baaad\")"
-    )]
-    fn record_service_consumed_panics_on_fatal_errors() {
+    fn accountant_schedule_retry_payable_scanner_because_not_all_pending_payables_completed() {
         init_test_logging();
-        let wallet = make_wallet("booga");
-        let payable_dao = PayableDaoMock::new().more_money_payable_result(Err(
-            PayableDaoError::RusqliteError("we cannot help ourselves; this is baaad".to_string()),
-        ));
-        let subject = AccountantBuilder::default()
-            .payable_daos(vec![ForAccountantBody(payable_dao)])
+        let test_name =
+            "accountant_schedule_retry_payable_scanner_because_not_all_pending_payables_completed";
+        let finish_scan_params_arc = Arc::new(Mutex::new(vec![]));
+        let retry_payable_notify_params_arc = Arc::new(Mutex::new(vec![]));
+        let mut subject = AccountantBuilder::default()
+            .logger(Logger::new(test_name))
             .build();
+        let pending_payable_scanner = ScannerMock::new()
+            .finish_scan_params(&finish_scan_params_arc)
+            .finish_scan_result(PendingPayableScanResult::PaymentRetryRequired);
+        subject
+            .scanners
+            .replace_scanner(ScannerReplacement::PendingPayable(ReplacementType::Mock(
+                pending_payable_scanner,
+            )));
+        subject.scan_schedulers.payable.retry_payable_notify =
+            Box::new(NotifyHandleMock::default().notify_params(&retry_payable_notify_params_arc));
+        let system = System::new(test_name);
+        let (mut msg, _) =
+            make_report_transaction_receipts_msg(TxStatus::Pending, TxStatus::Failed);
```

Comment:
If you could create a function that only outputs one `msg`, that would be ideal, then we can use it either once or multiple times as needed.